build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -585,8 +586,93 @@ export const handlers: Handler[] = [
             return { resolved: resolved };
         },
     },
+    {
+        name: "npm-package-json-prettier",
+        match,
+        handler: (file) => {
+            let json;
+
+            try {
+                json = JSON.parse(readFile(file));
+            } catch (err) {
+                return "Error parsing JSON file: " + file;
+            }
+
+            const hasPrettierScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier",
+            );
+            const hasPrettierFixScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier:fix",
+            );
+            const hasFormatScript = Object.prototype.hasOwnProperty.call(json.scripts, "format");
+
+            const missingScripts: string[] = [];
+
+            if (!hasPrettierScript) {
+                missingScripts.push(`prettier`);
+            }
+
+            if (!hasPrettierFixScript) {
+                missingScripts.push(`prettier:fix`);
+            }
+
+            if (!hasFormatScript) {
+                missingScripts.push(`format`);
+            }
+
+            return missingScripts.length > 0
+                ? `${file} is missing the following scripts: ${missingScripts.join("\n\t")}`
+                : undefined;
+        },
+        resolver: (file) => {
+            let json;
+            try {
+                json = JSON.parse(readFile(file));
+            } catch (err) {
+                return { resolved: false, message: "Error parsing JSON file: " + file };
+            }
+
+            const resolved = true;
+            addPrettier(json);
+            writeFile(file, JSON.stringify(sortPackageJson(json), undefined, 2) + newline);
+
+            return { resolved: resolved };
+        },
+    },
 ];
 
+function addPrettier(json: Record<string, any>) {
+    // if the json file contains scripts field
+    console.log(1);
+
+    const hasScriptsField = Object.prototype.hasOwnProperty.call(json, "scripts");
+
+    if (hasScriptsField) {
+        const hasPrettierScriptResolver = Object.prototype.hasOwnProperty.call(
```

Comment:
Is there an advantage to this way of doing the call over `json.scripts.hasOwnProperty("prettier")`? Or something like suggested in [this comment](https://github.com/microsoft/FluidFramework/pull/13072/files#r1036546558).

Reply:
I see. I think I like `json.scripts?.prettier !== undefined` better anyway 😄 .

And aren't you supposed to be OOF @tylerbutler ? Shoo! 😛